### ISSUE TITLE
Question: does a sniff exist to enforce this way of writing function multiline arguments

### ISSUE THREAD

**Issue by popovserhii** [squizlabs/PHP_CodeSniffer#2998]
These ones:
```php
return $this->render('LibraryItem:list.html.twig',
    [
        'library_items' => $libraryItems,
    ]
);

return $this->render(
    'LibraryItem:list.html.twig',
    [
        'library_items' => $libraryItems,
    ]
);

return $this->render('LibraryItem:list.html.twig', array(
        'library_items' => $libraryItems,
    )
);
```
Force to this:
```php
return $this->render('LibraryItem:list.html.twig', [
    'library_items' => $libraryItems,
]);
```
This is the short, simple and readable format, which is well described in [The PSR-2 Meta Document](https://www.php-fig.org/psr/psr-2/meta/#31---multi-line-arguments-09082013).

**Comment by gsherwood** [comment]
Short answer: what you are asking for is not described in PSR2 (see below for details) and so the included PSR2 standard and sniffs will not do exactly what you are asking for. They will not force all arguments to be defined on the same line, as you've requested. They enforce all PSR2 rules, which allow for more flexibility than you're looking for.

Your code samples have a mix of different issues in them.

As per PSR2, the first code sample is invalid because you have already split the argument list over multiple lines but started the first argument on the same line as the function call. The built-in PSR2 standard will ensure you start the first argument on the line after the function call. It will not enforce that all arguments need to be on the same line as this is not what PSR2 requires.

The second code sample is valid PSR2 code because you've manually placed arguments across multiple lines correctly.

The third code sample is not valid because you've got the the arguments on a single line, but you haven't placed the closing brace correctly when doing so.

If you use the built-in PSR2 standard, you will get the following errors for your code sample:
```
FILE: temp.php
---------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
---------------------------------------------------------------------------------------------------------
  2 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
 15 | ERROR | [x] Expected 0 spaces before closing parenthesis; newline found
---------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------
```
If you use PHPCBF to autofix the file, you'll get this:
```
<?php
return $this->render(
    'LibraryItem:list.html.twig',
    [
        'library_items' => $libraryItems,
    ]
);

return $this->render(
    'LibraryItem:list.html.twig',
    [
        'library_items' => $libraryItems,
    ]
);

return $this->render('LibraryItem:list.html.twig', array(
        'library_items' => $libraryItems,
    ));
```

**Comment by popovserhii** [comment]
@gsherwood 
> Short answer: what you are asking for is not described in PSR2

I don't understand this. Does this [PSR-2 Meta Document ](https://www.php-fig.org/psr/psr-2/meta/#3-errata) not including in PSR2, or what you mean?

I understand that this is correct according to PSR2:
```php
return $this->render(
    'LibraryItem:list.html.twig',
    [
        'library_items' => $libraryItems,
    ]
);
```
The reason is that I don't want to have different formats in our code even if they're valid according to PSR2 standard.

Maybe you know how can I write a rule that enforces the code to the needable format?

**Comment by gsherwood** [comment]
> I don't understand this. Does this [PSR-2 Meta Document ](https://www.php-fig.org/psr/psr-2/meta/#3-errata) not including in PSR2, or what you mean?

Yes, the meta document is part of PSR2, but it still allows developers to split argument lists over multiple lines. It just defines what this looks like for things like arrays and closures. You seem to want to ban splitting arguments over multiple lines, but this is not what PSR2 is trying to enforce.

> Maybe you know how can I write a rule that enforces the code to the needable format?

You'd have to write a custom sniff - none exist within PHP_CodeSniffer to force the syntax you are asking for. Unfortunately, I don't have any time to help with something like this.
